st2tests/st2tests/base.py

```diff
@@ -402,6 +403,81 @@ def assertProcessExited(self, proc):
             self.fail('Process with pid "%s" is still running' % (proc.pid))
 
 
+class BaseSensorTestCase(TestCase):
+    """
+    Base class for sensor tests.
+
+    This class provides some utility methods for verifying that a trigger has
+    been dispatched, etc.
+    """
+
+    sensor_cls = None
+
+    def setUp(self):
+        super(BaseSensorTestCase, self).setUp()
+
+        class_name = self.sensor_cls.__name__
+        sensor_wrapper = MockSensorWrapper(pack='tests', class_name=class_name)
+        self.sensor_service = MockSensorService(sensor_wrapper=sensor_wrapper)
+
+    def get_sensor_instance(self, config=None, poll_interval=None):
+        """
+        Retrieve instance of the sensor class.
+        """
+        kwargs = {
+            'sensor_service': self.sensor_service
+        }
+
+        if config:
+            kwargs['config'] = config
+
+        if poll_interval is not None:
+            kwargs['poll_interval'] = poll_interval
+
+        instance = self.sensor_cls(**kwargs)  # pylint: disable=not-callable
+        return instance
+
+    def get_dispatched_triggers(self):
+        return self.sensor_service.dispatched_triggers
+
+    def get_last_dispatched_trigger(self):
+        return self.sensor_service.dispatched_triggers[-1]
+
+    def assertTriggerDispatched(self, trigger, payload=None, trace_context=None):
+        """
+        Assert that the trigger with the provided values has been dispatched.
+
+        :param trigger: Name of the trigger.
+        :type trigger: ``str``
+
+        :param paylod: Trigger payload (optional). If not provided, only trigger name is matched.
+        type: payload: ``object``
+
+        :param trace_context: Trigger trace context (optional). If not provided, only trigger name
+                              is matched.
+        type: payload: ``object``
+        """
+        dispatched_triggers = self.get_dispatched_triggers()
+        for item in dispatched_triggers:
+            trigger_matches = (item['trigger'] == trigger)
+
+            if payload:
+                payload_matches = (item['payload'] == payload)
+            else:
+                payload_matches = True
+
+            if trace_context:
+                trace_context_matches = (item['trace_context'] == trace_context)
+            else:
+                trace_context_matches = True
+
+            if trigger_matches and payload_matches and trace_context_matches:
+                return True
+
+        msg = 'Trigger "%s" hasn\'t been dispatched' % (trigger)
+        raise AssertionError(msg)
```

Comment:
That's actually a bit weird:

```python======================================================================
FAIL: test_poll (test_sensor_docker_sensor.DockerSensorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/w/stackstorm/st2contrib/packs/docker/tests/test_sensor_docker_sensor.py", line 41, in test_poll
    payload={'container_info': MOCK_CONTAINER_DATA})
  File "/w/stackstorm/st2/st2tests/st2tests/base.py", line 477, in assertTriggerDispatched
    self.assertTrue(False, 'Trigger "%s" hasn\'t been dispatched' % (trigger))
AssertionError: False is not true : Trigger "docker.container_tracker.startedx" hasn't been dispatched
-------------------- >> begin captured logging << --------------------
````

```python
FAIL: test_poll (test_sensor_docker_sensor.DockerSensorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/w/stackstorm/st2contrib/packs/docker/tests/test_sensor_docker_sensor.py", line 41, in test_poll
    payload={'container_info': MOCK_CONTAINER_DATA})
  File "/w/stackstorm/st2/st2tests/st2tests/base.py", line 478, in assertTriggerDispatched
    raise AssertionError(msg)
AssertionError: Trigger "docker.container_tracker.startedx" hasn't been dispatched
-------------------- >> begin captured logging << --------------------
```

Imo, assertion error looks better and more appropriate.